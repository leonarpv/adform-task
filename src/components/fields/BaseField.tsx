import React, { ReactNode } from 'react'
import classNames from 'classnames'
import './baseField.scss'

export type BaseFieldProps = {
    children?: ReactNode
    label?: string
    error?: string
}

const BaseField = ({ children, label, error }: BaseFieldProps) => {
    const className = classNames('base-field', {
        'base-field--errored': error,
    })

    return (
        <div className={className}>
            {error && <span className="base-field__error-indicator" />}
            {label && <label className="base-field__label">{label}:</label>}
            {children}
            {error && <span className="base-field__error">{error}</span>}
        </div>
    )
}

export default BaseField
