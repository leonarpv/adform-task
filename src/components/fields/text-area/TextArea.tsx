import React, { DetailedHTMLProps, TextareaHTMLAttributes } from 'react'
import BaseField, { BaseFieldProps } from '../BaseField'
import './textArea.scss'

type Props = DetailedHTMLProps<
    TextareaHTMLAttributes<HTMLTextAreaElement>,
    HTMLTextAreaElement
> &
    BaseFieldProps

const TextArea = ({ label, error, ...rest }: Props) => {
    return (
        <BaseField label={label} error={error}>
            <textarea className="text-area" {...rest} />
        </BaseField>
    )
}

export default TextArea
