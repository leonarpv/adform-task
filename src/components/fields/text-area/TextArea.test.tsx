import { shallow } from 'enzyme'
import React from 'react'
import BaseField from '../BaseField'
import TextArea from './TextArea'

describe('TextArea', () => {
    it('should render properly', () => {
        const wrapper = shallow(<TextArea />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should render text area', () => {
        const wrapper = shallow(<TextArea />)

        expect(wrapper.find(BaseField).exists()).toBe(true)
        expect(wrapper.find('textarea.text-area').exists()).toBe(true)
    })
})
