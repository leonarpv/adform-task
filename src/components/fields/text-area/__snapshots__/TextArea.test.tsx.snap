// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextArea should render properly 1`] = `ShallowWrapper {}`;
